src/hooks.ts

```diff
@@ -34,29 +34,24 @@ export const createAPIHooks = <Endpoints extends RoughEndpoints>({
         options,
       );
     },
-    useAPIInfiniteQuery: (route, initPayload, options) => {
-      const { nextPageParamKey } = initPayload;
-      // @ts-expect-error, typescript enforcing only deleting non-required props
-      delete initPayload.nextPageParamKey;
+    useInfiniteAPIQuery: (route, initPayload, options) => {
       const queryKey: QueryKey = [createQueryKey(name, route, initPayload)];
-      return useInfiniteQuery(
+      const query = useInfiniteQuery(
         queryKey,
         ({ pageParam }) => {
-          const nextPayload =
-            pageParam && nextPageParamKey
-              ? {
-                  ...initPayload,
-                  [nextPageParamKey]: pageParam,
-                }
-              : undefined;
-          const payload = nextPayload || initPayload;
+          const payload = {
+            ...initPayload,
+            ...pageParam,
+          } as typeof initPayload;
```

Comment:
is the casting still needed here? Maybe we could explain why we need it in a comment.